lib/member_name_parts.rb

```diff
@@ -0,0 +1,37 @@
+# frozen_string_literal: true
+require 'scraped'
+
+class MemberNameParts < Scraped::HTML
+  field :prefix do
+    partitioned.first.join(' ')
+  end
+
+  field :name do
+    # .sub called for specific case where Dr. is not separated from name by space
+    partitioned.last.join(' ').sub('Dr.Bharrat', 'Bharrat')
```

Comment:
this really doesn't seem like the way to handle this. If the problem is that not all prefixes are followed by a space, then the fix is to change how we're extracting the prefixes, not to hard-code one specific case — that should always just be a very last resort.

Here I'd change `words` to be something more like:

```ruby
  noko.text.tidy.sub(/^Hon.? /, '').sub('.', '. ').split(/\s+/)
```

i.e. add a space after any full stop, before splitting on spaces.